### PR TITLE
Donut frosting

### PR DESCRIPTION
**Pull request by Luxizzle**
[feature]

## About the PR 

Adds donut frosting, first time using the icing tube on a donut will and a layer of icing and second time using the icing tube on a donut will add sprinkles

image

I've moved all the custom donuts to a seperate `.../donut/custom/...` path and made sure you cant add frosting to them.

Adding a frosting adds 15 reagents to the donut and a donut has a max reagent amount of 50, prefilled with the original 20 sugar. If this is too much i can alter it. Maybe we can block donuts from being filled by syringes too?

## Why's this needed? 

Because currently there is no way to add frosting to a donut and now you can get creative with it!

## Changelog

```changelog
(u)Luxizzle
(*)You can now frost donuts with an icing tube!
```

### CODE REVIEW

**Comment by Varshie** [comment]
I love this so much. More customizable foods would be a great addition to the chef role.

**Comment by nefarious6th** [comment]
I don't have much problem with being able to inject donuts--adds a fun element to potentially dangerous donuts. I think the only thing I'd maybe want altered is the reagent cap? Frosting + syringing + the initial sugar in the donut seems like a lot of capacity for perhaps more unsavory reagents/combos.
All in all, good PR, though! I've wanted to see something like this for donuts for a while! Maybe once chef XP is added this could be a way to earn some XP too!

**Comment by Luxizzle** [comment]
Maybe i wasnt clear in this, but the cap is at 50, adding a frosting layer adds 15, which you can do twice, once for the icing and once for the sprinkles. If you inject them with a syringe it can also hit the 50 cap. If the donut is too full it will refuse to add a frosting layer. So it shares the same maximum capacity.

File: code/modules/food_and_drink/snacks.dm [hunk 1]
```diff
@@ -1194,32 +1200,72 @@
...
+			initial_volume = 36
+			initial_reagents = list("sugar"=12,"synaptizine"=12,"epinephrine"=12)
+
+		random
+			New()
+				if(rand(1,3) == 1)
+					src.icon_state = "donut2"
+					src.name = "frosted donut"
+					src.heal_amt = 2
+				..()
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:1268]: weird code but out of scope

**Comment by nefarious6th** [comment]
> Maybe i wasnt clear in this, but the cap is at 50, adding a frosting layer adds 15, which you can do twice, once for the icing and once for the sprinkles. If you inject them with a syringe it can also hit the 50 cap. If the donut is too full it will refuse to add a frosting layer. So it shares the same maximum capacity.
> 
> So you can either add 30 extra reagents through the more superior frosting, using a syringe or a combination

Oh, then with that in mind, I don't really have a problem with the math for the frosting! Most of the chems I was thinking about for people injecting with a syringe are liquid anyways, which means they can be frosted on regardless, so I don't think there's a real problem then with syringe v. frosting if they don't stack reagents the way I thought! Thank you!

File: code/modules/food_and_drink/snacks.dm [hunk 2]
```diff
@@ -1194,32 +1200,72 @@
...
+		tube.reagents.trans_to(src, 15)
+		user.show_text("You add some [style] to [src]", "red")
+		src.style_step += 1
+
+	attackby(obj/item/I, mob/user)
+		if (istype(I, /obj/item/reagent_containers/food/drinks/drinkingglass/icing))
+			src.add_frosting(I, user)
+			return
+
+	custom
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:1237]:
```suggestion
	attackby(obj/item/I, mob/user)
		if (istype(I, /obj/item/reagent_containers/food/drinks/drinkingglass/icing))
			src.add_frosting(I, user)
			return
		else
			. = ..()

	custom
```

File: code/modules/food_and_drink/snacks.dm [hunk 3]
```diff
@@ -1179,12 +1179,18 @@
...
 	heal_amt = 1
-	initial_volume = 20
-	initial_reagents = "sugar"
+	initial_volume = 50
+	initial_reagents = list("sugar" = 20)
 	food_effects = list("food_energized")
+	var/can_add_frosting = TRUE
+	var/list/styles = list("icing", "sprinkles")
+	var/list/style_tags = list("icing" = "icing", "sprinkles" = "sprinkles")
+	var/list/style_step = 1
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:1193]: list?

File: code/modules/food_and_drink/snacks.dm [hunk 4]
```diff
@@ -1194,32 +1200,72 @@
...
+			user.show_text("The [tube] isn't full enough to add frosting.", "red")
+			return
+
+		if (src.style_step > length(src.styles))
+			user.show_text("You can't add anymore frosting.", "red")
+			return
+
+		// When a user also fills the donut with a syringe it can get a bit crowded in the donut.
+		if (src.reagents.is_full())
+			user.show_text("It feels like adding anything more to [src] would overfill it?", "red")
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:1218]:
```suggestion
			user.show_text("It feels like adding anything more to [src] would overfill it.", "red")
```

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Review by ZeWaka — APPROVED** [review]

File: code/modules/food_and_drink/snacks.dm [hunk 5]
```diff
@@ -1179,12 +1179,18 @@
...
+	appearance_flags = KEEP_TOGETHER
 	heal_amt = 1
-	initial_volume = 20
-	initial_reagents = "sugar"
+	initial_volume = 50
+	initial_reagents = list("sugar" = 20)
 	food_effects = list("food_energized")
+	var/can_add_frosting = TRUE
+	var/list/styles = list("icing", "sprinkles")
+	var/list/style_tags = list("icing" = "icing", "sprinkles" = "sprinkles")
```
Comment (flrsh) [inline code/modules/food_and_drink/snacks.dm:1192]: Wouldn't just one list suffice?

**Review by flrsh — CHANGES_REQUESTED** [review]
Very cute feature!

File: code/modules/food_and_drink/snacks.dm (hunk 5 above)
Comment (Luxizzle) [reply code/modules/food_and_drink/snacks.dm:1192]: This was more to differentiate between the icon name and what was displayed to the user originally, but you're right, that can be removed now.

**Review by flrsh — APPROVED** [review]

**Pull request merged**